front/lib/rate_limiter.ts

```diff
@@ -0,0 +1,30 @@
+import { v4 as uuidv4 } from "uuid";
+
+import { redisClient } from "./redis";
+
+export async function rateLimiter(
+  key: string,
+  maxPerTimeframe: number,
+  timeframeSeconds: number
+) {
```

Comment:
Can you log as a distribution the latency here so that we can keep track of the "cost" :pray:

Reply:
You mean log the time it took to get the remaining count? Because this would be exactly the redis latency. The complexity of this call is log(n) with n being the number of granted call that went through the rate limiter.
In these cases, it would at most log(30).

What could be interesting to log is  the distribution of "remaining" calls to fine tune the limits if needed.

Reply:
Oh I just want a monitor around this whole function to see how long it takes to run and whether it's stable or spiky